processor/transformprocessor/internal/common/functions.go

```diff
@@ -12,44 +12,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+package common
 
 import (
 	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/collector/model/pdata"
-
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
 )
 
 var registry = map[string]interface{}{
 	"keep_keys": keepKeys,
 	"set":       set,
 }
 
+type PathExpressionParser func(*Path) (GetSetter, error)
```

Comment:
The only real change in this PR besides moving around is the introduction of this type, which parses a path expression in a data-model specific way. The signals pass in the expression parser which allows the rest of the logic to apply generically.